polymorphic.h

```diff
@@ -404,7 +414,19 @@ class polymorphic {
     }
   }
 };
-
+#ifdef XYZ_HAS_EXTENDED_CONSTRUCTOR_TEMPLATE_ARGUMENT_DEDUCTION
+template <typename Value>
+polymorphic(Value) -> polymorphic<Value>;
+
+template <typename Alloc, typename Value,
+          typename std::enable_if_t<!is_polymorphic_v<Value>, int> = 0>
```

Comment:
```suggestion
template <typename Alloc, typename Value>
```
Same here: the deduction guide on line 427 is more specific so the `enable_if` isn't needed.

And a big bonus here: you can get rid of your `xyz::is_polymorphic_v` which is confusingly dissimilar to `std::is_polymorphic_v`.

Reply:
`std::is_polymorphic_v<std::polymorphic<T>>` evaluating to `false` is a pretty compelling call for a rename. A new issue shall be raised.

Reply:
FYI, `std::is_array_v<std::array<int, 10>>` and `std::is_function_v<std::function<int()>>` are both false too. (My [C++ Pub Quiz](https://quuxplusone.github.io/blog/2021/10/28/cppcon-2021-pub-quiz/) research is finally paying off! ;)) I think it's a _bonus_ to be able to _eliminate_ `xyz::is_polymorphic`, but if it's really indispensable, then I don't think you should worry about changing the name.